### PR TITLE
Pass the `amdgpu_target` and `cuda_arch` variants to DLAF dependencies

### LINKED ISSUE #684: Make DLA-Future spack dependencies inherit the same amdgpu_target values specified
Right now in spack, `amdgpu_target` should be specified for each dependencies of dla-future which is cumbersome.
Example:
https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/umpire/package.py#L111-L115

### PR DESCRIPTION
**Pull request by aurianer**
Fix #684

### CODE REVIEW

File: spack/packages/dla-future/package.py [hunk 1]
```diff
@@ -68,6 +68,19 @@ class DlaFuture(CMakePackage, CudaPackage, ROCmPackage):
...
+                when="amdgpu_target={0}".format(val))
+            depends_on("hipblas amdgpu_target={0}".format(val),
+                when="amdgpu_target={0}".format(val))
+            depends_on("rocsolver amdgpu_target={0}".format(val),
+                when="amdgpu_target={0}".format(val))
+            depends_on("rocblas amdgpu_target={0}".format(val),
+                when="amdgpu_target={0}".format(val))
+            depends_on("umpire amdgpu_target={0}".format(val),
+                when="amdgpu_target={0}".format(val))
+
```
Comment (msimberg) [inline spack/packages/dla-future/package.py:83]: I remember asking about this but I don't remember the answer: What will spack do if `dla-future` has `amdgpu_target=gfx90a` and e.g. `rocblas` has been installed with either `amdgpu_target=auto` or `amdgpu_target=gfx90a,more-targets...`?

Comment (msimberg) [reply spack/packages/dla-future/package.py:83]: Thanks! No I think if it works with one dependency it should work the same for any.

File: spack/packages/dla-future/package.py [hunk 2]
```diff
@@ -68,6 +68,19 @@ class DlaFuture(CMakePackage, CudaPackage, ROCmPackage):
 
     conflicts("+cuda", when="+rocm")
 
+    with when("+rocm"):
+        for val in ROCmPackage.amdgpu_targets:
+            depends_on("pika amdgpu_target={0}".format(val),
+                when="amdgpu_target={0}".format(val))
+            depends_on("hipblas amdgpu_target={0}".format(val),
+                when="amdgpu_target={0}".format(val))
```
Comment (msimberg) [inline spack/packages/dla-future/package.py:76]: It looks like `hipblas` doesn't actually have an `amdgpu_target` variant.

**Comment by msimberg** [comment]
Would you mind doing the same changes (including `CMAKE_HIP_ARCHITECTURES`) for CUDA (separately, if you so prefer)?

**Review by msimberg — APPROVED** [review]
Looks good to me, thanks!

**Comment by rasolca** [comment]
bors try

**Review by rasolca — APPROVED** [review]

**Comment by msimberg** [comment]
I restarted one of the build jobs which failed while pushing to jfrog. It's running here: https://gitlab.com/cscs-ci/eth-cscs/DLA-Future/-/jobs/3282678255. The changes in this PR should have no effect on that job.

**Pull request merged**